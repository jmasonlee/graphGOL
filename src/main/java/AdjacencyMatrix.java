import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;

public class AdjacencyMatrix {
    private ArrayList<ArrayList<Integer>> matrix = new ArrayList<>();

    public static final Integer DISCONNECTED = -1;

    public AdjacencyMatrix(){}

    public Integer distanceBetween(int nodeIndex1, int nodeIndex2) {
        return matrix.get(nodeIndex1).get(nodeIndex2);
    }

    public void addNode(int nodeIndex1) {
        ArrayList<Integer> edges = new ArrayList<Integer>(Collections.nCopies(matrix.size(), DISCONNECTED));
        matrix.add(edges);
        matrix.stream().forEach(n -> n.add( DISCONNECTED ));
    }

    public void createEdge(int nodeIndex1, int nodeIndex2, Integer distance) {
        matrix.get(nodeIndex1).set(nodeIndex2, distance);
    }

    public void addNodes(int[] nodeIndexes) {
        Arrays.stream(nodeIndexes).forEach(n -> addNode(n));
    }

    public void deleteNode(int nodeIndex) {
        matrix.remove(nodeIndex);
        matrix.stream().forEach(n -> n.remove(nodeIndex));
    }

    public int size(){
        return matrix.size();
    }

    public void deleteEdge(int nodeIndex1, int nodeIndex2) {
        changeEdge(nodeIndex1, nodeIndex2, DISCONNECTED);
    }

    public void changeEdge(int nodeIndex1, int nodeIndex2, Integer newDistance) {
        matrix.get(nodeIndex1).set(nodeIndex2, newDistance);
        matrix.get(nodeIndex2).set(nodeIndex1, newDistance);
    }
}
