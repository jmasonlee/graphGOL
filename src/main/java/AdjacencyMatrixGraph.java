import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;

public class AdjacencyMatrixGraph {
    private ArrayList<ArrayList<Integer>> matrix = new ArrayList<>();

    public AdjacencyMatrixGraph(){}

    public Integer distanceBetween(Integer node1, Integer node2) {
        return matrix.get(node1).get(node2);
    }

    public void addNode(Integer node1) {
        ArrayList<Integer> edges = new ArrayList<Integer>(Collections.nCopies(matrix.size(), -1));
        matrix.add(edges);
        matrix.stream().forEach(n -> n.add( -1));
    }

    public void createEdge(Integer node1, Integer node2, Integer distance) {
        matrix.get(node1).set(node2, distance);
    }

    public void addNodes(Integer[] nodes) {
        Arrays.stream(nodes).forEach(n -> addNode(n));
    }

//    public void deleteNode(Integer node) {
//        matrix.remove(node);
//        matrix.stream().forEach(n -> n.remove(node));
//    }

    public int size(){
        return matrix.size();
    }
}
