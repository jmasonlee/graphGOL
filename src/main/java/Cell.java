public class Cell implements Comparable {
    public Integer y;
    public Integer x;

    public Cell(Integer x, Integer y) {
        this.x = x;
        this.y = y;
    }

    @Override
    public int compareTo(Object o) {
        return 0;
    }
}
