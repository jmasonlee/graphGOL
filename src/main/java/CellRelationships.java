public enum CellRelationships {
    DISCONNECTED, NEIGHBOUR, COPARENT
}
