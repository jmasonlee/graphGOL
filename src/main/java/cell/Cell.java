package cell;

public class Cell implements Comparable {
  public Integer y;
  public Integer x;

  public Cell(Integer x, Integer y) {
    this.x = x;
    this.y = y;
  }

  @Override
  public int compareTo(Object o) {
    return 0;
  }

  @Override
  public boolean equals(Object o) {

    if (o == this) {
      return true;
    }

    if (!(o instanceof Cell)) {
      return false;
    }

    Cell otherCell = (Cell) o;

    return otherCell.x == this.x && otherCell.y == this.y;
  }

  @Override
  public String toString(){
      return "[" + x + "," + y + "]";
  }
}
