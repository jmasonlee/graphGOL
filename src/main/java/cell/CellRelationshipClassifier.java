package cell;

public class CellRelationshipClassifier {
    public static Relationships classify(Cell cell, Cell cell1) {
        return Relationships.SELF;
    }

    public enum Relationships {
        DISCONNECTED(-1), SELF(0), NEIGHBOUR(1), COPARENT(2);

        public final int maxDistance;

        private Relationships(int maxDistance){
            this.maxDistance = maxDistance;
        }
    }


}
