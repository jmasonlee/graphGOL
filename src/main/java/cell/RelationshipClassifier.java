package cell;

public class RelationshipClassifier {
    public static Relationships classify(Cell cell, Cell cell1) {
        return Relationships.SELF;
    }

    public enum Relationships {
        DISCONNECTED(-1), SELF(0), NEIGHBOUR(1), COPARENT(2);

        public final int maxDistance;

        private Relationships(int maxDistance){
            this.maxDistance = maxDistance;
        }
    }


}
