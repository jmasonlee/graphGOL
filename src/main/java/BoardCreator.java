import cell.Cell;
import org.apache.commons.lang.StringUtils;

import java.util.List;

public class BoardCreator {

    public static String createBoardOutput(List<Cell> cells){
        StringBuilder emptyBoard = emptyBoardOfSize(getBoardWidth(), 5);
        return populateBoardWithCells(emptyBoard, cells);
    }

    private static String populateBoardWithCells(StringBuilder emptyBoard, List<Cell> cells) {
        StringBuilder populatedBoard = emptyBoard;
        cells.forEach(c -> {
            Integer positionInBoard = getCellPositionInBoard(c);
            populatedBoard.setCharAt(positionInBoard, 'X');
        });
        return populatedBoard.toString();
    }

    private static Integer getCellPositionInBoard(Cell cell) {
        return (cell.x * getBoardWidth())+ cell.x + cell.y;
    }

    private static Integer getBoardWidth() {
        return 5;
    }

    private static StringBuilder emptyBoardOfSize(int width, int height) {
        StringBuilder board = new StringBuilder();

        for (int i = 0; i < height; i++) {
            board.append(StringUtils.repeat(" ", width));
            if(i < height-1){
                board.append("\n");
            }
        }

        return board;
    }
}