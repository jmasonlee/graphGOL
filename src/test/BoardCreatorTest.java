import cell.Cell;
import org.approvaltests.Approvals;
import org.junit.Test;

import java.util.*;

public class BoardCreatorTest {
  @Test
  public void testBoardIsGeneratedWithCellsInCorrectLocation() {
      List<Integer> coordinateNumbers = Arrays.asList(0, 1, 2, 3, 4);
      String[] allBoards = new String[25];


      for(int x = 0; x < coordinateNumbers.size(); x++){
          for(int y = 0; y < coordinateNumbers.size(); y++){
              int index = x*5 + y;
              String input = String.format("(%d,%d):\n",x,y);
              allBoards[index] = input + BoardCreator.createBoardOutput(Arrays.asList(new Cell(x,y)));
          }
      }

      Approvals.verifyAll("",allBoards);
  }

}