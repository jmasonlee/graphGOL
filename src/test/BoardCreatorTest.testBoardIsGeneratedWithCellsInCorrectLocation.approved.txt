[00] = (0,0):
X    
     
     
     
     

[01] = (0,1):
 X   
     
     
     
     

[02] = (0,2):
  X  
     
     
     
     

[03] = (0,3):
   X 
     
     
     
     

[04] = (0,4):
    X
     
     
     
     

[05] = (1,0):
     
X    
     
     
     

[06] = (1,1):
     
 X   
     
     
     

[07] = (1,2):
     
  X  
     
     
     

[08] = (1,3):
     
   X 
     
     
     

[09] = (1,4):
     
    X
     
     
     

[10] = (2,0):
     
     
X    
     
     

[11] = (2,1):
     
     
 X   
     
     

[12] = (2,2):
     
     
  X  
     
     

[13] = (2,3):
     
     
   X 
     
     

[14] = (2,4):
     
     
    X
     
     

[15] = (3,0):
     
     
     
X    
     

[16] = (3,1):
     
     
     
 X   
     

[17] = (3,2):
     
     
     
  X  
     

[18] = (3,3):
     
     
     
   X 
     

[19] = (3,4):
     
     
     
    X
     

[20] = (4,0):
     
     
     
     
X    

[21] = (4,1):
     
     
     
     
 X   

[22] = (4,2):
     
     
     
     
  X  

[23] = (4,3):
     
     
     
     
   X 

[24] = (4,4):
     
     
     
     
    X

